lib/kontrast/api_client.rb

```diff
@@ -0,0 +1,56 @@
+require 'faraday'
+
+module Kontrast
+    class ApiClient
+
+        attr_reader :responses, :env
+        attr_writer :headers
+
+        def initialize(env, host, app_id, app_secret, headers: {})
+            @env = env
+            @host = host
+            @app_id = app_id
+            @app_secret = app_secret
+            @connection = nil
+            @responses = {}
+            @headers = headers
+        end
+
+        def fetch(path, save_file: false, folder_name: "")
+            response = connection.get("#{path}") do |req|
+                req.headers['Authorization'] = "Bearer #{token}"
+                req.headers.merge!(@headers)
+            end
+            data = JSON.parse(response.body)
```

Comment:
If this is to be used generically, we can't always expect a JSON response. We should check that the `Content-Type` response header actually is JSON and pretty-print it to a file in that case.